python/paddle/v2/fluid/tests/book/test_rnn_encoder_decoder.py

```diff
@@ -196,9 +196,51 @@ def main():
             print('pass_id=' + str(pass_id) + ' batch=' + str(batch_id) +
                   " avg_cost=" + str(avg_cost_val))
             if batch_id > 3:
+                if save_dirname is not None:
+                    fluid.io.save_inference_model(
+                        save_dirname, ['source_sequence',
+                                       'target_sequence'], [prediction], exe)
```

Comment:
I think we need to rethink about the decoding for inference. In this example, we take the `target_embedding`as input in the `lstm_decoder_without_attention()`, which requires the target_sequence. However, while doing inference, ideally we don't have access to the target, hence, the decoding while inference has to be modified.

Reply:
I agree. We can refine the inference example in future PR.